src/main/java/org/example/service/jdbc/TimetableService.java

```
@@ -0,0 +1,184 @@
+            groupsHasTTEntriesDAO.createEntity(groupsHasTimetableEntryToAdd);
+            logger.debug(String.format("Timetable entry (%d) assigned to group (%d) to the service", ttEntryId, groupId));
+            return groupsHasTimetableEntryToAdd;
+        } else {
+            logger.error("Timetable entry couldn't be assigned to group in the service as one of its id is invalid");
+            throw new NullPointerException("Timetable entry or student group id is invalid - can't assign it in service");
+        }
+    }
+
+    @Override
+    public boolean removeTimetableEntryFromGroup(long ttEntryId, long groupId) {
+        if (ttEntryId > 0 && groupId > 0) {
+            int result = groupsHasTTEntriesDAO.removeEntityById(groupId, ttEntryId);
+            if (result == 1) {
+                logger.debug(String.format("Timetable entry (%d) assigned to group (%d) from the service", ttEntryId, groupId));
+                return true;
+            } else {
+                logger.error(String.format("Timetable entry (%d) couldn't be assigned to group (%d) in the the service",
+                        ttEntryId, groupId));
+                return false;
+            }
+        } else {
+            logger.error("Timetable entry couldn't be assigned to group in the service as one of its id is invalid");
+            return false;
+        }
+    }
+
+    @Override
+    public List<TimetableEntry> getTimetableForStudentGroup(long groupId) throws EntityNotFoundException {
+        List<TimetableEntry> timetableByGroupId = new ArrayList<>();
+        if (groupId > 0) {
+            List<Long> ids = groupsHasTTEntriesDAO.getTimetableEntryIdsByGroupId(groupId);
+            for (long id : ids) {
+                timetableByGroupId.add(getTimetableEntryById(id));
+            }
+        } else {
+            logger.error("Time table entries assigned to group couldn't be retrieved from the service as timetable entry " +
```

maybe you better throw an exception here instead of returning an empty list.